FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/abstractions/platform/freertos/transport_interface_freertos.c

```diff
@@ -0,0 +1,151 @@
+                                       uint16_t port,
+                                       uint32_t receiveTimeoutMs )
+{
+    Socket_t tcpSocket = FREERTOS_INVALID_SOCKET;
+    BaseType_t socketStatus = 0;
+    struct freertos_sockaddr serverAddress = { 0 };
+    TickType_t receiveTimeout = pdMS_TO_TICKS( receiveTimeoutMs );
+
+    /* Create a new TCP socket. */
+    tcpSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_STREAM, FREERTOS_IPPROTO_TCP );
+
+    if( tcpSocket == FREERTOS_INVALID_SOCKET )
+    {
+        LogError( ( "Failed to create new socket." ) );
+        socketStatus = -1;
+    }
+    else
+    {
+        LogDebug( ( "Created new TCP socket." ) );
+        socketStatus = FreeRTOS_setsockopt( tcpSocket,
+                                            0,
+                                            FREERTOS_SO_RCVTIMEO,
+                                            &receiveTimeout,
+                                            sizeof( TickType_t ) );
+
+        if( socketStatus != 0 )
+        {
+            LogError( ( "Failed to set socket receive timeout." ) );
+        }
+    }
+
+    if( socketStatus == 0 )
+    {
+        /* Establish connection. */
+        serverAddress.sin_family = FREERTOS_AF_INET;
+        serverAddress.sin_port = FreeRTOS_htons( port );
+        serverAddress.sin_addr = FreeRTOS_gethostbyname( pHostName );
+        serverAddress.sin_len = ( uint8_t ) sizeof( serverAddress );
+
+        /* Check for errors from DNS lookup. */
+        if( serverAddress.sin_addr == 0 )
+        {
+            LogError( ( "Failed to resolve %s.", pHostName ) );
+            socketStatus = -1;
+        }
+    }
+
+    if( socketStatus == 0 )
+    {
+        socketStatus = FreeRTOS_connect( tcpSocket, &serverAddress, sizeof( serverAddress ) );
+    }
+
+    /* Clean up on failure. */
+    if( socketStatus != 0 )
+    {
+        if( tcpSocket != FREERTOS_INVALID_SOCKET )
+        {
+            FreeRTOS_closesocket( tcpSocket );
+        }
+    }
+    else
+    {
+        /* Set the socket. */
+        pNetworkContext->tcpSocket = tcpSocket;
+        LogDebug( ( "TCP Connection to %s established.", pHostName ) );
+    }
+
+    return socketStatus;
+}
+
+void Transport_FreeRTOS_Disconnect( const NetworkContext_t * pNetworkContext )
+{
+    BaseType_t socketStatus;
+
+    if( pNetworkContext->tcpSocket != FREERTOS_INVALID_SOCKET )
+    {
+        socketStatus = FreeRTOS_shutdown( pNetworkContext->tcpSocket, FREERTOS_SHUT_RDWR );
+        FreeRTOS_closesocket( pNetworkContext->tcpSocket );
+    }
+}
+
+int32_t Transport_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
+                                 void * pBuffer,
+                                 size_t bytesToRecv )
+{
+    int32_t socketStatus = 0;
+
+    socketStatus = FreeRTOS_recv( pNetworkContext->tcpSocket, pBuffer, bytesToRecv, 0 );
+
+    /* If no data is available, transport receive should return 0. */
+    if( socketStatus == FREERTOS_EWOULDBLOCK )
```

Comment:
Timeout already returns 0 as per the documentation here https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/API/recv.html. "If a time out occurred before data could be received then 0 is returned"